cwrap/cfile.py

```diff
@@ -14,57 +14,147 @@
+            """
+            c_ptr = self._as_file(py_file)
+            print("ASDASDASDASDA")
+            print(py_file, type(py_file), c_ptr)
+            print("ASDASDASDASDA")
+            try:
+                super(CFILE, self).__init__(c_ptr)
+            except ValueError:
+                raise ValueError("Sorry - the supplied argument is not a valid "
+                                " Python file handle!")
+
+            self.py_file = py_file
+
+        def __del__(self):
+            pass
 
-        The purpose of the CFILE class is to be able to use python
-        file handles when calling C functions which expect a FILE
-        pointer. A CFILE instance should be created based on the
-        Python file handle, and that should be passed to the function
-        expecting a FILE pointer.
 
-        The implementation is based on the ctypes object
-        pythonapi which is ctypes wrapping of the CPython api.
+if six.PY3:
+    from .clib import load as cwrapload
 
-          C-function:
-             void fprintf_hello(FILE * stream , const char * msg);
 
-          Python wrapper:
-             lib = clib.load( "lib.so" )
-             fprintf_hello = Prototype(lib, "void fprintf_hello( FILE , char* )")
+    class LibcPrototype(Prototype):
+        lib = cwrapload('libc', '.6')
 
-          Python use:
-             py_fileH = open("file.txt" , "w")
-             fprintf_hello( CFILE( py_fileH ) , "Message ...")
-             py_fileH.close()
+        def __init__(self, prototype, bind=False):
+            super(LibcPrototype, self).__init__(
+                LibcPrototype.lib, prototype, bind=bind)
 
-        If the supplied argument is not of type py_file the function
-        will raise a TypeException.
 
-        Examples: ecl.ecl.ecl_kw.EclKW.fprintf_grdecl()
+    class CWrapFile(BaseCClass):
         """
-        c_ptr = self._as_file(py_file)
-        try:
-            super(CFILE, self).__init__(c_ptr)
-        except ValueError as e:
-            raise TypeError("Sorry - the supplied argument is not a valid Python file handle!")
+        TODO
+        """
+
+        TYPE_NAME = "FILE"
+
+        _fopen = LibcPrototype("void* fopen (char*, char*)")
+        _fclose = LibcPrototype("int fclose (FILE)", bind=True)
+        _fflush = LibcPrototype("int fflush (FILE)", bind=True)
+
+        def __init__(self, fname, mode):
+            c_ptr = self._fopen(fname, mode)
+            self._mode = mode
+            self._fname = fname
+            self._closed = False
+
+            try:
+                super(CWrapFile, self).__init__(c_ptr)
+                print(dir(self))
+            except ValueError:
+                self._closed = True
+                raise IOError('Could not open file "{}" in mode {}'
+                              .format(fname, mode))
+
+        def close(self):
+            if not self.closed:
+                print(self)
+                self._fflush()
```

Comment:
The `fclose()` will flush the user buffers to the OS - no need to explicitly call `fflush()`